fluent/client/client.go

```diff
@@ -113,50 +114,84 @@ func New(opts ConnectionOptions) *Client {
 
 // Connect initializes the Session and Connection objects by opening
 // a client connect to the target configured in the ConnectionFactory
-func (c *Client) Connect() error {
+func (c *Client) Session() *Session {
+	c.sessionLock.RLock()
+	defer c.sessionLock.RUnlock()
+
+	return c.session
+}
+
+func (c *Client) connect() error {
 	conn, err := c.New()
 	if err != nil {
 		return err
 	}
 
-	c.Session = &Session{
+	c.session = &Session{
 		Connection: conn,
 	}
 
 	// If no shared key, handshake mode is not required
 	if c.AuthInfo.SharedKey == nil {
-		c.Session.TransportPhase = true
+		c.session.TransportPhase = true
 	}
 
 	return nil
 }
 
-func (c *Client) Reconnect() error {
-	panic("not implemented yet")
+// Connect initializes the Session and Connection objects by opening
+// a client connect to the target configured in the ConnectionFactory
+func (c *Client) Connect() error {
+	c.sessionLock.Lock()
+	defer c.sessionLock.Unlock()
+
+	if c.session != nil {
+		return errors.New("a session is already active")
+	}
+
+	return c.connect()
 }
 
-// Disconnect terminates a client connection
-func (c *Client) Disconnect() (err error) {
-	if c.Session != nil {
-		if c.Session.Connection != nil {
-			err = c.Session.Connection.Close()
-		}
+func (c *Client) disconnect() (err error) {
+	if c.session != nil && c.session.Connection != nil {
```

Comment:
Why here the check for `c.session.Connection != nil` and not in Session.Connect() where `c.session != nil` seems sufficient?